aerogear/mobile-security-service
# Refactor ui & redux

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8922
https://issues.jboss.org/browse/AEROGEAR-8938

## What
Full refactor of UI and redux

## Why
Better management of UI components and best practices


## Verification Steps
- verify there are no failing tests
  - `npm test`
- verify test coverage is > 70%
  - `npm run coverage`
- verify there are no regressions in functionality
  - check applications view
    - check header 
    - click on an app version
  - check application versions view
    - check a version can be enabled and disabled
    - check all versions can be disabled, and a custom message added for all versions
    - check a custom message for a specific version can be added, and also updated
    - check navigation back via breadcrumb works
  - _list of regressions not conclusive - please poke around & check everything works as you would expect_

## Checklist:


## Progress


## Additional Notes
